lib/elixir/lib/calendar/date.ex

```diff
@@ -100,7 +110,8 @@ defmodule Date do
 
   """
   @doc since: "1.5.0"
-  @spec range(Calendar.date(), Calendar.date()) :: Date.Range.t()
+  @spec range(Calendar.date(), Calendar.date() | Duration.t() | [duration_unit_pair]) ::
```

Comment:
I think this would improve docs (now they would still show `range(first, last)`:

```suggestion
  @spec range(start :: Calendar.date(), end_or_duration :: Calendar.date() | Duration.t() | [duration_unit_pair]) ::
```

Reply:
while I like `start`/`end`, should we stick to `first`/`last` so it's:

```
@spec range(first :: Calendar.date(), last_or_duration :: Calendar.date() | Duration.t() | [duration_unit_pair]) ::
```

Reply:
![Screenshot 2025-01-13 at 10 27 29](https://github.com/user-attachments/assets/d4fdec5f-abe9-4f51-af02-fa4db3c57ad9)

![Screenshot 2025-01-13 at 10 30 34](https://github.com/user-attachments/assets/2900f441-09d7-42c5-b5ba-d832684221c6)

Reply:
`last_or_duration` is ok but the "duration" part doesn't work as well with `first` IMO, compared to `start` + `duration`/`start` + `end`. Either way not a big deal 🙃